test/e2e/network/service.go

```diff
@@ -2309,6 +2311,466 @@ var _ = common.SIGDescribe("Services", func() {
 		}
 	})
 
+	ginkgo.It("should fail health check node port if there are only terminating endpoints [Feature:ProxyTerminatingEndpoints]", func() {
+		// windows kube-proxy does not support this feature yet
+		e2eskipper.SkipIfNodeOSDistroIs("windows")
+
+		// This behavior is not supported if Kube-proxy is in "userspace" mode.
+		// So we check the kube-proxy mode and skip this test if that's the case.
+		if proxyMode, err := proxyMode(f); err == nil {
+			if proxyMode == "userspace" {
+				e2eskipper.Skipf("The test doesn't work with kube-proxy in userspace mode")
+			}
+		} else {
+			framework.Logf("Couldn't detect KubeProxy mode - test failure may be expected: %v", err)
+		}
```

Comment:
userspace has been deprecated so we don't need this, also we should try to make the tests avoiding assumptions on kube-proxy so other implementations can benefit

Reply:
We do this elsewhere. We can get rid of it all when we actually remove the userspace proxy.

Reply:
I think it's a fair ask to clean this up, I'll follow-up with an issue or a PR on this

Reply:
Sounds good